solr/connection.go

```diff
@@ -138,9 +138,27 @@ func (c *Connection) SetBasicAuth(username, password string) {
 	c.password = password
 }
 
+func (c *Connection) AddHeader(key, value string) {
+	header := []string{key, value}
+	c.headers = append(c.headers, header)
+}
+
 func (c *Connection) Resource(source string, params *url.Values) (*[]byte, error) {
 	params.Set("wt", "json")
-	r, err := HTTPGet(fmt.Sprintf("%s/%s/%s?%s", c.url.String(), c.core, source, params.Encode()), nil, c.username, c.password)
+	baseUrl := fmt.Sprintf("%s/%s/%s", c.url.String(), c.core, source)
+	encodedParameters := params.Encode()
+	maximumUrlLength := 2083
```

Comment:
I think this a limitation in browser not solr?

Reply:
I just realised that it would be nice to be able to adjust this value in case some one wants to allow higher value. How about put it as exported var?

Reply:
It is a limitation of http protocol.
Solr follow the general advice given in [RFC7230](https://tools.ietf.org/html/rfc7230#section-3.1.1): "Various ad hoc limitations on request-line length are found in practice. It is RECOMMENDED that all HTTP senders and recipients support, at a minimum, request-line lengths of 8000 octets."
This [stackoverflow answer](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers) suggest that search engines like URLs < 2048 chars

Reply:
I think a exported var works fine in this case.